src/TLuaInterpreter.cpp

```diff
@@ -17968,154 +17968,72 @@ int TLuaInterpreter::getScroll(lua_State* L)
-        lua_pushboolean(L, host.mEnableMSP);
-        return 1;
-    }
-
-    if (key == qsl("askTlsAvailable")) {
-        lua_pushboolean(L, host.mAskTlsAvailable);
-        return 1;
-    }
-
-    if (key == qsl("inputLineStrictUnixEndings")) {
-        lua_pushboolean(L, host.mUSE_UNIX_EOL);
-        return 1;
-    }
-
-    if (key == qsl("autoClearInputLine")) {
-        lua_pushboolean(L, host.mAutoClearCommandLineAfterSend);
-        return 1;
-    }
-
-    if (key == qsl("showSentText")) {
-        lua_pushboolean(L, host.mPrintCommand);
-        return 1;
-    }
-
-    if (key == qsl("fixUnnecessaryLinebreaks")) {
-        lua_pushboolean(L, host.mUSE_IRE_DRIVER_BUGFIX);
-        return 1;
-    }
-
-    if (key == qsl("specialForceCompressionOff")) {
-        lua_pushboolean(L, host.mFORCE_NO_COMPRESSION);
-        return 1;
-    }
-
-    if (key == qsl("specialForceGAOff")) {
-        lua_pushboolean(L, host.mFORCE_GA_OFF);
-        return 1;
-    }
-
-    if (key == qsl("specialForceCharsetNegotiationOff")) {
-        lua_pushboolean(L, host.mFORCE_CHARSET_NEGOTIATION_OFF);
-        return 1;
-    }
-
-    if (key == qsl("specialForceMxpNegotiationOff")) {
-        lua_pushboolean(L, host.mFORCE_MXP_NEGOTIATION_OFF);
-        return 1;
-    }
-
-    if (key == qsl("compactInputLine")) {
-        lua_pushboolean(L, host.getCompactInputLine());
-        return 1;
-    }
-
-    if (key == qsl("announceIncomingText")) {
-        lua_pushboolean(L, host.mAnnounceIncomingText);
-        return 1;
-    }
-
-    if (key == qsl("blankLinesBehaviour")) {
-        const auto behaviour = host.mBlankLineBehaviour;
-        if (behaviour == Host::BlankLineBehaviour::Show) {
-            lua_pushstring(L, "show");
-        } else if (behaviour == Host::BlankLineBehaviour::Hide) {
-            lua_pushstring(L, "hide");
-        } else if (behaviour == Host::BlankLineBehaviour::ReplaceWithSpace) {
-            lua_pushstring(L, "replacewithspace");
-        }
-        return 1;
-    }
+        return warnArgumentValue(L, __func__, "you must provide a key");
+    }
+
+    static const std::unordered_map<QString, std::function<void(lua_State*, Host&)>> configMap = {
+        { qsl("mapRoomSize"), [](lua_State* L, Host& host){ lua_pushnumber(L, host.mRoomSize); } },
+        { qsl("mapExitSize"), [](lua_State* L, Host& host){ lua_pushnumber(L, host.mLineSize); } },
+        { qsl("mapRoundRooms"), [](lua_State* L, Host& host){ lua_pushboolean(L, host.mBubbleMode); } },
+        { qsl("showRoomIdsOnMap"), [](lua_State* L, Host& host){ lua_pushboolean(L, host.mShowRoomID); } },
+        { qsl("show3dMapView"), [](lua_State* L, Host& host){
+            #if defined(INCLUDE_3DMAPPER)
+            if (host.mpMap && host.mpMap->mpMapper) {
+                auto widget = host.mpMap->mpMapper->glWidget;
+                lua_pushboolean(L, (widget && widget->isVisible()));
+                return;
+            }
+            #endif
```

Comment:
IIRC we do NOT indent `#thing`s:
```suggestion
#if defined(INCLUDE_3DMAPPER)
            if (host.mpMap && host.mpMap->mpMapper) {
                auto widget = host.mpMap->mpMapper->glWidget;
                lua_pushboolean(L, (widget && widget->isVisible()));
                return;
            }
#endif
```